CAIPs/caip-x.md

```diff
@@ -0,0 +1,99 @@
+  
+### Specification
+
+**Extension ID Announcement Protocol:**
+
+Each wallet extension must implement a mechanism to announce its extension ID to a dApp. This should be done using a JavaScript event emitted by the wallet and listened for by the dApp. The event should contain the canonical extension ID as its payload.
+
+**Modification to dApp Listening Mechanisms:**
+
+dApps must modify their initialization routines to listen for the wallet's extension ID announcement event. Upon capturing the event, the dApp can store the extension ID and use it to establish a connection using the `externally_connectable` API.
+
+**Content Script Utilization**:
+
+While this CAIP significantly advances the way wallets interact with dApps through the `externally_connectable` API, it still necessitates the use of a content script. This script, however, requires fewer permissions compared to traditional methods and can be injected synchronously to facilitate the initial communication process. This adjustment is necessary to bridge the current browser limitations and set the stage for future enhancements.
+
+1.  **Reduced Permission Scope**: The content script deployed under this new standard does not need the extensive permissions typically required for full page content access and manipulation. Instead, its role is limited to emitting and listening for the specific event that carries the wallet's extension ID, thereby reducing the risk and scope of permissions required.
+    
+2.  **Synchronous Injection**: The content script can be injected synchronously with the page load, ensuring that the wallet's extension ID is available as soon as the dApp begins its initialization process. This method avoids the delays and potential conflicts associated with asynchronous script loading, leading to a smoother and more predictable interaction pattern.
+    
+3.  **Security and Efficiency**: Although using a content script still presents a minimal footprint on the user's browsing experience, this approach is a compromise that balances current browser capabilities with the need for enhanced security and performance. By minimizing the script's functionality to only what is necessary for ID transmission, the risks associated with broader access permissions are mitigated.
+    
+
+It worth acknowledging the transitional nature of the proposed solution, aiming for a future where browsers might offer more direct APIs for wallet capability registration and discovery, eliminating the need for any content script altogether. Meanwhile, the approach detailed in this CAIP ensures a more secure and efficient method of wallet-to-dApp communication than previously employed techniques, setting a foundation for further innovations in this space.
+
+ **Example Implementation:**
+
+- Wallets:
+
+```javascript
+
+document.dispatchEvent(new CustomEvent('walletExtensionIdAnnouncement', {
+
+detail: {
+
+extensionId: 'your_extension_id_here'
```

Comment:
i believe we should also include the name, app ID, and maybe icon

Reply:
What is an app ID?

Reply:
ahh sorry, i meant rdns 

https://eips.ethereum.org/EIPS/eip-6963#rdns